cpp/ql/test/library-tests/dataflow/fields/partial-definition-diff.ql

```diff
@@ -5,43 +5,14 @@
 import cpp
 import semmle.code.cpp.ir.dataflow.DataFlow::DataFlow as IR
 import semmle.code.cpp.dataflow.DataFlow::DataFlow as AST
+import Nodes
 
-newtype TNode =
-  TASTNode(AST::Node n) or
-  TIRNode(IR::Node n)
-
-class Node extends TNode {
-  string toString() { none() }
-
-  IR::Node asIR() { none() }
-
-  AST::Node asAST() { none() }
-
-  Location getLocation() { none() }
-}
-
-class ASTNode extends Node, TASTNode {
-  AST::Node n;
-
-  ASTNode() { this = TASTNode(n) }
-
+class ASTPartialDefNode extends ASTNode {
   override string toString() { result = n.asPartialDefinition().toString() }
-
-  override AST::Node asAST() { result = n }
-
-  override Location getLocation() { result = n.getLocation() }
 }
 
-class IRNode extends Node, TIRNode {
-  IR::Node n;
-
-  IRNode() { this = TIRNode(n) }
-
+class IRPartialDefNode extends IRNode {
   override string toString() { result = n.asPartialDefinition().toString() }
-
-  override IR::Node asIR() { result = n }
-
-  override Location getLocation() { result = n.getLocation() }
 }
```

Comment:
Should these have a charpred? I know it's fine in the current version and we don't edit `.ql` test files much, but it seems like the sort of thing that could break subtly in the future.

Reply:
Good idea. I've added a charpred now. Is that the one you had in mind?

Reply:
Yep